index.js

```diff
@@ -512,57 +514,82 @@ function buildInnerObject (schema, name, externalSchema, fullSchema) {
   return { code: code, laterCode: laterCode }
 }
 
-function buildObject (schema, code, name, externalSchema, fullSchema) {
+function addIfThenElse (schema, name, externalSchema, fullSchema) {
+  var code = ''
+  var r
+  var laterCode = ''
+  var innerR
+
+  const i = schema.if
+  const then = schema.then
+  const e = schema.else
+  delete schema.if
+  delete schema.then
+  delete schema.else
```

Comment:
we might want to clone this before deleting.

Reply:
why?

Reply:
Good point I clone the schema deeply

Reply:
done